tests/query/parser/test_conditions.py

```diff
@@ -0,0 +1,338 @@
+import pytest
+
+from typing import Any, Sequence
+
+from snuba.datasets.factory import get_dataset
+from snuba.query.expressions import (
+    Column,
+    Expression,
+    FunctionCall,
+    Lambda,
+    Literal,
+    Argument,
+)
+from snuba.query.conditions import (
+    ConditionFunctions,
+    BooleanFunctions,
+)
+from snuba.query.parser.conditions import parse_conditions_to_expr
+from snuba.util import tuplify
+
+test_conditions = [
+    ([], None,),
+    ([[[]], []], None,),
```

Comment:
I think this should be a parse error, but apparently today it is supported so I am preserving the behavior.